includes/core/server/request/class-create-intention.php

```diff
@@ -194,6 +194,8 @@ public function set_shipping( array $shipping ) {
 	 * @return mixed           Either the same response, or the correct object.
 	 */
 	public function format_response( $response ) {
+		$this->api_client->update_customer_currencies( $response );
```

Comment:
Should we hook this into a `woocommerce_thankyou` action, for example?

Reply:
I tested this action with different flows and found that it only works when the payment is completed (ha!), despite the documentation stating that it should also be triggered by the `on hold`, `pending`, `failed`, and `cancelled` statuses. I tested the capture later flow and the action was not triggered.

Since there is an advanced filter on the analytics page that allows the merchant to filter by any status, I believe we should update the customer currencies option even if the order is created manually.

<table>
<tr>
<td>
<img width="500" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/16882226/e02693cf-10a8-4385-8630-804ffb2deefe">
</td>
<td>
<img width="500" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/16882226/d8678350-33ad-4f06-b8aa-7db467654cd8">
</td>
</tr>
<tr>
<td>
<i>Advanced filters</i>
</td>
<td>
<i>Available statuses</i>
</td>
</tr>
</table>

After testing several available actions, I found that `woocommerce_order_status_changed` made the most sense to me. This is because there are multiple entry points to the payment flow, and the action also supports manually created orders.

<table>
<tr>
<td>
<img src="https://github.com/Automattic/woocommerce-payments/assets/16882226/8fa09424-d703-43fb-883b-eb171ce330de" />
</td>
</tr>
<tr>
<td>
<a href="https://woocommerce.com/document/managing-orders/#visual-diagram-to-illustrate-order-statuses"><i>Visual Diagram to Illustrate Order Statuses</i></a>
</td>
</tr>
</table>

It was updated on https://github.com/Automattic/woocommerce-payments/pull/7143/commits/1b5bf554a57bf08b336d0fb62168143fbb15a207. Thanks for the suggestion, and let me know your thoughts on this!